elide-core/src/main/java/com/yahoo/elide/core/ErrorObjects.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2018, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Error Objects, see http://jsonapi.org/format/#error-objects. <br><br>
+ * Builder example: <br>
+ * <pre><code>
+ *   ErrorObjects errorObjects = ErrorObjects.create()
+ *       .addError()
+ *           .withDetail("first error message")
+ *       .addError()
+ *           .withDetail("second error message")
+ *       .build();
+ * </code></pre>
+ */
+public class ErrorObjects {
+    @Getter private final List<Map<String, Object>> errors;
+
+    public ErrorObjects(List<Map<String, Object>> errors) {
+        this.errors = Objects.requireNonNull(errors, "errors must not be null");
+    }
+
+    public static ErrorObjectsBuilder create() {
+        return new ErrorObjectsBuilder(new ErrorObjects(new ArrayList<>()));
+    }
+
+    /**
+     * ErrorObjectsBuilder.
+     */
+    public static class ErrorObjectsBuilder {
+
+        private final ErrorObjects container;
+        private final List<Map<String, Object>> errors;
+
+        ErrorObjectsBuilder(ErrorObjects container) {
+            this.container = container;
+            this.errors = container.getErrors();
+        }
+
+        public ErrorObjectBuilder addError() {
+            Map<String, Object> errorObject = new HashMap<>();
+            errors.add(errorObject);
+            return new ErrorObjectBuilder(this, errorObject);
+        }
+
+        public ErrorObjects build() {
+            if (errors.isEmpty()) {
+                throw new IllegalArgumentException("at least one error object");
+            }
+            return container;
+        }
+
+        /**
+         * ErrorObjectBuilder.
+         */
+        public static class ErrorObjectBuilder {
```

Comment:
I appreciate how your two builders chain together nicely, but that makes me think that they're really just two parts of the same logical unit. I think we should combine them and just have the `ErrorObjectsBuilder` keep a pointer to the error we are currently building.